src/synapse/azext_synapse/_params.py

```diff
@@ -0,0 +1,161 @@
+
+    for scope in ['create', 'update']:
+        with self.argument_context('synapse workspace ' + scope) as c:
+            c.argument('sql_admin_login_password', help='The sql administrator login password.')
+            c.argument('tags', arg_type=tags_type, help='The tag of workspace.')
+            c.argument('identity_type', help='The type of managed identity.')
+
+    with self.argument_context('synapse workspace create') as c:
+        c.argument("account_url", help='The data lake storage account url.')
+        c.argument('file_system', help='The file system of the data lake storage account.')
+        c.argument('sql_admin_login_user', help='The sql administrator login user name.')
+
+    with self.argument_context('synapse workspace update') as c:
+        c.argument('principal_id', help='The principal id of managed identity.')
+
+    # synapse spark pool
+    with self.argument_context('synapse spark pool') as c:
+        c.argument('workspace_name', help='The name of the workspace.')
+
+    for scope in ['show', 'create', 'update', 'delete']:
+        with self.argument_context('synapse spark pool ' + scope) as c:
+            c.argument('spark_pool_name', arg_type=name_type, help='The name of the spark pool.')
+
+    with self.argument_context('synapse spark pool create') as c:
+        # Node
+        c.argument('node_count', arg_group='Node', help='The number of node.')
+        c.argument('node_size_family', arg_group='Node', help='The node size family.')
+        c.argument('node_size', arg_group='Node', help='The node size.')
+
+        # AutoScale
+        c.argument('auto_scale_enabled', arg_type=get_three_state_flag(), arg_group='AutoScale',
+                   help='The flag of enabling auto scale.')
+        c.argument('max_node_count', arg_group='AutoScale', help='The max node count.')
+        c.argument('min_node_count', arg_group='AutoScale', help='The min node count.')
+
+        # AutoPause
+        c.argument('auto_pause_enabled', arg_type=get_three_state_flag(), arg_group='AutoPause',
+                   help='The flag of enabling auto pause.')
+        c.argument('delay_in_minutes', arg_group='AutoPause', help='The delay time.')
+
+        # Environment Configuration
+        c.argument('library_requirements_content', arg_group='Environment Configuration',
+                   help='The library requirements content.')
+        c.argument('library_requirements_filename', arg_group='Environment Configuration',
+                   help='The library requirements file name.')
+
+        # Default Folder
+        c.argument('spark_events_folder', arg_group='Default Folder', help='The spark events folder.')
+        c.argument('default_spark_log_folder', arg_group='Default Folder', help='The default spark log folder.')
+
+        # Component Version
+        c.argument('spark_version', arg_group='Component Version', help='The spark version.')

```

Comment:
>The spark version. [](start = 73, length = 18)

We should propose the supported versions here.

Reply:
> > The spark version. [](start = 73, length = 18)
> 
> We should propose the supported versions here.

If we propose the version, we still need to fix the code when the version is deprecated

Reply:
> > > The spark version. [](start = 73, length = 18)
> > 
> > 
> > We should propose the supported versions here.
> 
> If we propose the version, we still need to fix the code when the version is deprecated

updated